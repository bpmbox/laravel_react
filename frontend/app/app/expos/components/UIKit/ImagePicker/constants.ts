/**
 * Shared constants used by the image picker on all platforms.
 */

export const RESIZE_MAX_SIZE = 1024;
