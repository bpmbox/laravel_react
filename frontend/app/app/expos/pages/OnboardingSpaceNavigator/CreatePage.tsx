import { withNavigation } from '@react-navigation/core';
import { Formik, FormikActions, FormikProps } from 'formik';
import React, { ReactNode, useState } from 'react';
import { useTranslation } from 'react-i18next';
import { Keyboard, View } from 'react-native';
import * as yup from 'yup';
import ImagePicker from '../../components/UIKit/ImagePicker';
import Button from '../../components/UIKit/items/Button';
import SectionHeading from '../../components/UIKit/items/SectionHeading';
import SingleLineInput from '../../components/UIKit/items/SingleLineInput';
import Spacer from '../../components/UIKit/items/Spacer';
import Text from '../../components/UIKit/items/Text';
import Page from '../../components/UIKit/Layout/Page';
import i18n from '../../i18n';
import { slugBlacklist } from '../../libs/space-validators/slug-blacklist';
import { PARAM_SPACE } from '../../constants';
import routes from '../../routes';
import { ItemWidth } from '../../theme.style';
import spaceService from '../../services/space';
import AuthStore from '../../store/auth';

const CreatePage = (props: { navigation: any; }) => {
    const { navigation } = props;
    const { t } = useTranslation('CreatePage');
    const {currentUser} = AuthStore.useContainer();
    const [slugError, setSlugError] = useState<string | null>(null);

    const validationSchema = yup.object().shape({
        spaceIcon: yup.string(),
        spaceName: yup.string().required(t`Space name is required.`),
        spaceUrl: yup.string()
            .lowercase()
            .matches(/^([\w-_])+$/, t`Only valid URL characters allowed.`)
            .notOneOf(slugBlacklist),
    });

    type SpaceFormValues = {
        spaceName: string;
        spaceUrl?: string;
        spaceIcon?: string;
    };

    const handleSubmit = async (values: SpaceFormValues, actions: FormikActions<SpaceFormValues>): Promise<void> => {
        setSlugError(null);
        Keyboard.dismiss();
        try {
            const space = await spaceService.createSpace(values.spaceName, values.spaceUrl, values.spaceIcon);

            // Redirect to created space
            navigation.navigate(routes.MAIN_SPACE_REDIRECT, { [PARAM_SPACE]: space });
        } catch (e) {
            // TODO: use Error code.
            if (e.message === 'SpaceSlugAlreadyExists') {
                setSlugError(t`Already taken.`)
            } else {
                console.error(e);
                setSlugError(t`Unable to validate your space url.`);
            }
        } finally {
            actions.setSubmitting(false);
        }
    };

    return (
        <Page scrollable>
            <Formik
                validationSchema={validationSchema}
                initialValues={{spaceName: '', spaceUrl: ''}}
                onSubmit={handleSubmit}
            >
                {(formikProps: FormikProps<SpaceFormValues>): ReactNode => {
                    const {
                        values,
                        handleSubmit,
                        isValid,
                        isSubmitting,
                        handleChange,
                        handleBlur,
                        setFieldValue
                    } = formikProps;

                    return (
                        <View>
                            <Text
                                text={t`A space is where your community comes to life. You can create or join other spaces later.`}
                                light
                                small
                                desktopWidth={ItemWidth.narrow}
                                desktopCenterItem={true}
                                />
                            <ImagePicker
                                allowSelection
                                space
                                includeButton
                                buttonTitle={t`Set Logo`}
                                desktopWidth={ItemWidth.narrow}
                                desktopCenterItem={true}
                                onUploaded={(fileUrl: string) => {
                                    setFieldValue('spaceIcon', fileUrl);
                                }}/>
                            <SectionHeading
                                text={t`Space name`}
                                desktopWidth={ItemWidth.narrow}
                                desktopCenterItem={true} />
                            <SingleLineInput
                                testID={'SpaceNameInput'}
                                onChangeText={handleChange('spaceName')}
                                onBlur={handleBlur('spaceName')}
                                value={values.spaceName}
                                placeholder={t('E.g. company name')}
                                autoFocus
                                returnKeyType='next'
                                autoCorrect={false}
                                spellCheck={false}
                                autoCapitalize='words'
                                desktopWidth={ItemWidth.narrow}
                                desktopCenterItem={true} />
                            <SectionHeading
                                text={t`Space domain`}
                                desktopWidth={ItemWidth.narrow}
                                desktopCenterItem={true}/>
                            <SingleLineInput
                                testID={'SpaceUrlInput'}
                                onChangeText={handleChange('spaceUrl')}
                                onBlur={handleBlur('spaceUrl')}
                                value={values.spaceUrl}
                                inputLegend='withtree.com/'
                                placeholder={t('URL')}
                                returnKeyType='go'
                                autoCorrect={false}
                                spellCheck={false}
                                autoCapitalize='none'
                                desktopWidth={ItemWidth.narrow}
                                desktopCenterItem={true}/>

                            { slugError  &&
                                <Text
                                    text={slugError}
                                    danger
                                    small
                                    desktopWidth={ItemWidth.narrow}
                                    desktopCenterItem={true} />
                            }
                            <Spacer />
                            <Text
                                text={t`Invite people to join the space by sharing this link.`}
                                light
                                small
                                desktopWidth={ItemWidth.narrow}
                                desktopCenterItem={true}/>
                            <Spacer />
                            <Button
                                testID={'SubmitButton'}
                                onPress={() => handleSubmit()}
                                disabled={!isValid || isSubmitting}
                                text={t('Create Space')}
                                desktopWidth={ItemWidth.narrow}
                                desktopCenterItem={true}
                            />

                            {currentUser && currentUser.email &&
                                (
                                <>
                                    <Spacer />
                                    <Text
                                        text={t('Using account: {{email}}', { email: currentUser.email })}
                                        light
                                        small
                                        desktopWidth={ItemWidth.narrow}
                                        desktopCenterItem={true}
                                        />
                                </>
                                )
                            }
                        </View>
                    )
                }}
            </Formik>
        </Page>
    );
};

CreatePage.navigationOptions = {
    title: i18n.t('CreatePage::Create a Space'),
}

// @ts-ignore
CreatePage.path = 'create'; //override path for better web URLs

export default withNavigation(CreatePage);
