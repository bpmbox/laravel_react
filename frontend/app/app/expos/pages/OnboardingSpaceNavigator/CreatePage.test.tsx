import { act } from '@testing-library/react-hooks';
import React from 'react';
import TestRenderer from 'react-test-renderer';
import sinon, { SinonSandbox } from 'sinon';
import { PARAM_SPACE } from '../../constants';
import routes from '../../routes';
import spaceService from '../../services/space';
import AuthStore from '../../store/auth';
import CreatePage from './CreatePage';

describe('CreatePage', () => {
    let sandbox: SinonSandbox;
    beforeEach((): void => {
        sandbox = sinon.createSandbox();
        jest.resetModules();
    });

    afterEach((): void => {
        sandbox.restore();
    });

    it('should create space and redirect to it', async () => {
        const fakeNavigation = {
            navigate: jest.fn(),
        };

        const createSpace = sandbox.stub(spaceService, 'createSpace').resolves({
            id: 'space1',
            name: 'Winterfell',
            slug: 'winterfell',
            iconUrl: null,
        });

        const testRenderer = TestRenderer.create(
            <AuthStore.Provider
                initialState={
                    {
                        isAuthenticated: true,
                        isLoading: false,
                        currentUser: {
                            id: 'abc',
                            givenName: 'Jon',
                            familyName: 'Snow',
                            email: 'jonsnow@winterfell.com',
                            avatarUrl: null,
                        },
                    } as any
                }>
                <CreatePage navigation={fakeNavigation} />
            </AuthStore.Provider>
        );
        // Await initial render
        await act(async () => {
            await new Promise(setImmediate);
        });
        // Using snapshot to assert preconditions.
        expect(testRenderer.toJSON()).toMatchSnapshot();

        // Fill out form.
        await act(async () => {
            const spaceNameInput = testRenderer.root.findByProps({
                testID: 'SpaceNameInput',
            });
            spaceNameInput.props.onChangeText({
                persist: (): void => { },
                target: {
                    name: 'spaceName',
                    value: 'Winterfell',
                },
            });
            spaceNameInput.props.onBlur();
            await new Promise(setImmediate);
        });

        await act(async () => {
            const spaceUrlInput = testRenderer.root.findByProps({
                testID: 'SpaceUrlInput',
            });
            spaceUrlInput.props.onChangeText({
                persist: (): void => { },
                target: {
                    name: 'spaceUrl',
                    value: 'winterfell',
                },
            });
            spaceUrlInput.props.onBlur();
            await new Promise(setImmediate);
        });


        await act(async () => {
            const submitButton = testRenderer.root.findByProps({
                testID: 'SubmitButton',
            });
            submitButton.props.onPress();            
            await new Promise(setImmediate);
        });

        await act(async () => {
            await new Promise(setImmediate);
        });

        // Assert expected results.
        expect(createSpace.calledOnce).toBeTruthy();
        expect(createSpace.calledWith('Winterfell', 'winterfell')).toBeTruthy();
        // assert redirect occurrs.
        expect(fakeNavigation.navigate).toBeCalledWith(
            routes.MAIN_SPACE_REDIRECT, {
            [PARAM_SPACE]: {
                iconUrl: null,
                id: 'space1',
                name: 'Winterfell',
                slug: 'winterfell',
            }
        });
    });
});