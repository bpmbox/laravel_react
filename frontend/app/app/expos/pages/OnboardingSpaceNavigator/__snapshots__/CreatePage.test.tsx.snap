// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatePage should create space and redirect to it 1`] = `
<SafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <ScrollView
    contentContainerStyle={Array []}
    keyboardDismissMode="on-drag"
    keyboardShouldPersistTaps="handled"
    scrollable={true}
    style={
      Array [
        undefined,
      ]
    }
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "maxWidth": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "flex": 1,
              "flexDirection": "column",
              "flexGrow": 1,
              "justifyContent": "center",
              "maxWidth": "100%",
              "overflow": "hidden",
            }
          }
        >
          <Text
            selectable={false}
            style={
              Object {
                "color": "#777777",
                "fontFamily": "GenEiGothicP-SL",
                "fontSize": 14.875,
                "fontWeight": "400",
                "lineHeight": 22.3125,
                "maxWidth": "100%",
                "opacity": 1,
                "paddingBottom": 4,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 4,
                "textAlign": "left",
              }
            }
          >
            A space is where your community comes to life. You can create or join other spaces later.
          </Text>
        </View>
      </View>
      <View
        padded={true}
        style={
          Array [
            Object {},
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
            undefined,
            Object {},
            Object {},
            Object {
              "paddingHorizontal": 16,
              "paddingVertical": 16,
            },
          ]
        }
      >
        <TouchableOpacity
          activeOpacity={0.9}
          onPress={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#F6F6F6",
                "borderColor": "#EAEAEA",
                "borderRadius": 7.4375,
                "borderWidth": 0.5,
                "height": 100,
                "justifyContent": "center",
                "width": 100,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "height": 40,
                    "width": 40,
                  },
                ]
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight={40}
                bbWidth={40}
                fill="none"
                focusable={false}
                height={40}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    undefined,
                    Object {
                      "opacity": 1,
                    },
                    Object {
                      "flex": 0,
                      "height": 40,
                      "width": 40,
                    },
                  ]
                }
                vbHeight={40}
                vbWidth={40}
                width={40}
                xml="<svg width=\\"40\\" height=\\"40\\" viewBox=\\"0 0 40 40\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\">
<g clip-path=\\"url(#clip0)\\">
<path fill-rule=\\"evenodd\\" clip-rule=\\"evenodd\\" d=\\"M16.8 9.33331C16.4433 9.33331 16.1103 9.51155 15.9124 9.8083L14.0958 12.5333H10.4C9.55126 12.5333 8.73733 12.8705 8.13721 13.4706C7.53709 14.0707 7.19995 14.8846 7.19995 15.7333V27.4666C7.19995 28.3153 7.53709 29.1293 8.13721 29.7294C8.73733 30.3295 9.55126 30.6666 10.4 30.6666H29.6C30.4486 30.6666 31.2626 30.3295 31.8627 29.7294C32.4628 29.1293 32.7999 28.3153 32.7999 27.4666V15.7333C32.7999 14.8846 32.4628 14.0707 31.8627 13.4706C31.2626 12.8705 30.4486 12.5333 29.6 12.5333H25.9041L24.0875 9.8083C23.8896 9.51155 23.5566 9.33331 23.2 9.33331H16.8ZM24.2666 21.0667C24.2666 23.4231 22.3564 25.3333 20 25.3333C17.6436 25.3333 15.7333 23.4231 15.7333 21.0667C15.7333 18.7102 17.6436 16.8 20 16.8C22.3564 16.8 24.2666 18.7102 24.2666 21.0667Z\\" fill=\\"#CCCCCC\\"/>
</g>
<defs>
<clipPath id=\\"clip0\\">
<rect x=\\"7.19995\\" y=\\"7.20001\\" width=\\"25.6\\" height=\\"25.6\\" fill=\\"white\\"/>
</clipPath>
</defs>
</svg>
"
                xmlns="http://www.w3.org/2000/svg"
              >
                <RNSVGGroup
                  fill={null}
                  fillOpacity={1}
                  fillRule={1}
                  font={Object {}}
                  matrix={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                  opacity={1}
                  originX={0}
                  originY={0}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                  rotation={0}
                  scaleX={1}
                  scaleY={1}
                  skewX={0}
                  skewY={0}
                  stroke={null}
                  strokeDasharray={null}
                  strokeDashoffset={null}
                  strokeLinecap={0}
                  strokeLinejoin={0}
                  strokeMiterlimit={4}
                  strokeOpacity={1}
                  strokeWidth={1}
                  vectorEffect={0}
                  x={0}
                  y={0}
                >
                  <RNSVGGroup
                    clipPath="clip0"
                    fill={
                      Array [
                        0,
                        4278190080,
                      ]
                    }
                    fillOpacity={1}
                    fillRule={1}
                    font={Object {}}
                    matrix={
                      Array [
                        1,
                        0,
                        0,
                        1,
                        0,
                        0,
                      ]
                    }
                    opacity={1}
                    originX={0}
                    originY={0}
                    propList={Array []}
                    rotation={0}
                    scaleX={1}
                    scaleY={1}
                    skewX={0}
                    skewY={0}
                    stroke={null}
                    strokeDasharray={null}
                    strokeDashoffset={null}
                    strokeLinecap={0}
                    strokeLinejoin={0}
                    strokeMiterlimit={4}
                    strokeOpacity={1}
                    strokeWidth={1}
                    vectorEffect={0}
                    x={0}
                    y={0}
                  >
                    <RNSVGPath
                      d="M16.8 9.33331C16.4433 9.33331 16.1103 9.51155 15.9124 9.8083L14.0958 12.5333H10.4C9.55126 12.5333 8.73733 12.8705 8.13721 13.4706C7.53709 14.0707 7.19995 14.8846 7.19995 15.7333V27.4666C7.19995 28.3153 7.53709 29.1293 8.13721 29.7294C8.73733 30.3295 9.55126 30.6666 10.4 30.6666H29.6C30.4486 30.6666 31.2626 30.3295 31.8627 29.7294C32.4628 29.1293 32.7999 28.3153 32.7999 27.4666V15.7333C32.7999 14.8846 32.4628 14.0707 31.8627 13.4706C31.2626 12.8705 30.4486 12.5333 29.6 12.5333H25.9041L24.0875 9.8083C23.8896 9.51155 23.5566 9.33331 23.2 9.33331H16.8ZM24.2666 21.0667C24.2666 23.4231 22.3564 25.3333 20 25.3333C17.6436 25.3333 15.7333 23.4231 15.7333 21.0667C15.7333 18.7102 17.6436 16.8 20 16.8C22.3564 16.8 24.2666 18.7102 24.2666 21.0667Z"
                      fill={
                        Array [
                          0,
                          4291611852,
                        ]
                      }
                      fillOpacity={1}
                      fillRule={0}
                      matrix={
                        Array [
                          1,
                          0,
                          0,
                          1,
                          0,
                          0,
                        ]
                      }
                      opacity={1}
                      originX={0}
                      originY={0}
                      propList={
                        Array [
                          "fill",
                          "fillRule",
                        ]
                      }
                      rotation={0}
                      scaleX={1}
                      scaleY={1}
                      skewX={0}
                      skewY={0}
                      stroke={null}
                      strokeDasharray={null}
                      strokeDashoffset={null}
                      strokeLinecap={0}
                      strokeLinejoin={0}
                      strokeMiterlimit={4}
                      strokeOpacity={1}
                      strokeWidth={1}
                      vectorEffect={0}
                      x={0}
                      y={0}
                    />
                  </RNSVGGroup>
                  <RNSVGDefs>
                    <RNSVGClipPath
                      fill={
                        Array [
                          0,
                          4278190080,
                        ]
                      }
                      fillOpacity={1}
                      fillRule={1}
                      matrix={
                        Array [
                          1,
                          0,
                          0,
                          1,
                          0,
                          0,
                        ]
                      }
                      name="clip0"
                      opacity={1}
                      originX={0}
                      originY={0}
                      propList={Array []}
                      rotation={0}
                      scaleX={1}
                      scaleY={1}
                      skewX={0}
                      skewY={0}
                      stroke={null}
                      strokeDasharray={null}
                      strokeDashoffset={null}
                      strokeLinecap={0}
                      strokeLinejoin={0}
                      strokeMiterlimit={4}
                      strokeOpacity={1}
                      strokeWidth={1}
                      vectorEffect={0}
                      x={0}
                      y={0}
                    >
                      <RNSVGRect
                        fill={
                          Array [
                            0,
                            4294967295,
                          ]
                        }
                        fillOpacity={1}
                        fillRule={1}
                        height={25.6}
                        matrix={
                          Array [
                            1,
                            0,
                            0,
                            1,
                            0,
                            0,
                          ]
                        }
                        opacity={1}
                        originX={0}
                        originY={0}
                        propList={
                          Array [
                            "fill",
                          ]
                        }
                        rotation={0}
                        scaleX={1}
                        scaleY={1}
                        skewX={0}
                        skewY={0}
                        stroke={null}
                        strokeDasharray={null}
                        strokeDashoffset={null}
                        strokeLinecap={0}
                        strokeLinejoin={0}
                        strokeMiterlimit={4}
                        strokeOpacity={1}
                        strokeWidth={1}
                        vectorEffect={0}
                        width={25.6}
                        x={7.19995}
                        y={7.20001}
                      />
                    </RNSVGClipPath>
                  </RNSVGDefs>
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
          </View>
        </TouchableOpacity>
        <View
          style={
            Object {
              "width": 16,
            }
          }
        />
        <View>
          <View
            size={0}
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
                "width": "100%",
              }
            }
            title="Set Logo"
          >
            <TouchableOpacity
              activeOpacity={0.9}
              hoverStyle={
                Object {
                  "borderRadius": 7.4375,
                  "opacity": 1,
                  "shadowColor": "#000000",
                  "shadowOffset": Object {
                    "height": 10,
                    "width": 0,
                  },
                  "shadowOpacity": 0.1,
                  "shadowRadius": 20,
                }
              }
              onPress={[Function]}
              onPressIn={[Function]}
              onPressOut={[Function]}
            >
              <Text
                ellipsizeMode="tail"
                numberOfLines={1}
                style={
                  Object {
                    "backgroundColor": "#0066FF",
                    "borderColor": "#00000000",
                    "borderRadius": 7.4375,
                    "borderWidth": 0,
                    "color": "#FFFFFF",
                    "fontFamily": "GenEiGothicP-R",
                    "fontSize": 14.875,
                    "fontWeight": "500",
                    "lineHeight": 34,
                    "overflow": "hidden",
                    "paddingHorizontal": 16,
                    "textAlign": "center",
                    "width": "100%",
                  }
                }
              >
                Set Logo
              </Text>
            </TouchableOpacity>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "maxWidth": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "flex": 1,
              "flexDirection": "column",
              "flexGrow": 1,
              "justifyContent": "center",
              "maxWidth": "100%",
              "overflow": "hidden",
            }
          }
        >
          <Text
            selectable={false}
            style={
              Object {
                "color": "#000000",
                "fontFamily": "GenEiGothicP-B",
                "fontSize": 17,
                "fontWeight": "700",
                "lineHeight": 27.200000000000003,
                "maxWidth": "100%",
                "opacity": 1,
                "paddingBottom": 4,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 16,
                "textAlign": "left",
              }
            }
          >
            Space name
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 44,
            "maxWidth": "100%",
          }
        }
        testID="SpaceNameInput"
      >
        <View
          style={
            Object {
              "backgroundColor": "#EAEAEA",
              "bottom": 0,
              "height": 0.5,
              "left": 16,
              "position": "absolute",
              "right": 16,
            }
          }
        />
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "flex": 1,
              "flexDirection": "column",
              "flexGrow": 1,
              "justifyContent": "center",
              "maxWidth": "100%",
              "overflow": "hidden",
            }
          }
        >
          <Text
            ellipsizeMode="tail"
            numberOfLines={1}
            selectable={false}
            style={
              Object {
                "color": "#000000",
                "fontFamily": "GenEiGothicP-SL",
                "fontSize": 17,
                "fontWeight": "400",
                "lineHeight": 25.5,
                "maxWidth": "100%",
                "opacity": 1,
                "paddingBottom": 4,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 4,
                "textAlign": "left",
              }
            }
          >
            
          </Text>
        </View>
        <TextInput
          allowFontScaling={true}
          autoCapitalize="words"
          autoCorrect={false}
          autoFocus={true}
          editable={true}
          keyboardType="default"
          multiline={false}
          numberOfLines={1}
          onBlur={[Function]}
          onChangeText={[Function]}
          placeholder="E.g. company name"
          placeholderTextColor="#CCCCCC"
          rejectResponderTermination={true}
          returnKeyType="next"
          selectionColor="#0066FF"
          spellCheck={false}
          style={
            Object {
              "color": "#000000",
              "fontFamily": "GenEiGothicP-SL",
              "fontSize": 17,
              "fontWeight": "400",
              "minHeight": 44,
              "paddingBottom": 0,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 0,
              "width": "100%",
            }
          }
          textAlignVertical="top"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "maxWidth": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "flex": 1,
              "flexDirection": "column",
              "flexGrow": 1,
              "justifyContent": "center",
              "maxWidth": "100%",
              "overflow": "hidden",
            }
          }
        >
          <Text
            selectable={false}
            style={
              Object {
                "color": "#000000",
                "fontFamily": "GenEiGothicP-B",
                "fontSize": 17,
                "fontWeight": "700",
                "lineHeight": 27.200000000000003,
                "maxWidth": "100%",
                "opacity": 1,
                "paddingBottom": 4,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 16,
                "textAlign": "left",
              }
            }
          >
            Space domain
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 44,
            "maxWidth": "100%",
          }
        }
        testID="SpaceUrlInput"
      >
        <View
          style={
            Object {
              "backgroundColor": "#EAEAEA",
              "bottom": 0,
              "height": 0.5,
              "left": 16,
              "position": "absolute",
              "right": 16,
            }
          }
        />
        <View
          style={
            Array [
              Object {},
              Object {
                "alignItems": "center",
                "flexDirection": "row",
              },
              Object {
                "width": "100%",
              },
              Object {},
              Object {},
              Object {
                "paddingHorizontal": 0,
                "paddingVertical": 0,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "GenEiGothicP-SL",
                  "fontSize": 17,
                  "fontWeight": "400",
                  "lineHeight": 25.5,
                  "maxWidth": "100%",
                  "opacity": 1,
                  "paddingBottom": 4,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 4,
                  "textAlign": "left",
                },
                Object {
                  "lineHeight": 35,
                  "minHeight": 44,
                  "paddingRight": 10,
                },
              ]
            }
          >
            withtree.com/
          </Text>
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            autoCorrect={false}
            editable={true}
            keyboardType="default"
            multiline={false}
            numberOfLines={1}
            onBlur={[Function]}
            onChangeText={[Function]}
            placeholder="URL"
            placeholderTextColor="#CCCCCC"
            rejectResponderTermination={true}
            returnKeyType="go"
            selectionColor="#0066FF"
            spellCheck={false}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "GenEiGothicP-SL",
                  "fontSize": 17,
                  "fontWeight": "400",
                  "minHeight": 44,
                  "paddingBottom": 0,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 0,
                  "width": "100%",
                },
                Object {
                  "flex": 1,
                  "paddingLeft": 0,
                },
              ]
            }
            textAlignVertical="top"
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 44,
            "maxWidth": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "flex": 1,
              "flexDirection": "column",
              "flexGrow": 1,
              "justifyContent": "center",
              "maxWidth": "100%",
              "overflow": "hidden",
            }
          }
        >
          <Text
            ellipsizeMode="tail"
            numberOfLines={1}
            selectable={false}
            style={
              Object {
                "color": "#000000",
                "fontFamily": "GenEiGothicP-SL",
                "fontSize": 17,
                "fontWeight": "400",
                "lineHeight": 25.5,
                "maxWidth": "100%",
                "opacity": 1,
                "paddingBottom": 4,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 4,
                "textAlign": "left",
              }
            }
          >
            
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "maxWidth": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "flex": 1,
              "flexDirection": "column",
              "flexGrow": 1,
              "justifyContent": "center",
              "maxWidth": "100%",
              "overflow": "hidden",
            }
          }
        >
          <Text
            selectable={false}
            style={
              Object {
                "color": "#777777",
                "fontFamily": "GenEiGothicP-SL",
                "fontSize": 14.875,
                "fontWeight": "400",
                "lineHeight": 22.3125,
                "maxWidth": "100%",
                "opacity": 1,
                "paddingBottom": 4,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 4,
                "textAlign": "left",
              }
            }
          >
            Invite people to join the space by sharing this link.
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 44,
            "maxWidth": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "flex": 1,
              "flexDirection": "column",
              "flexGrow": 1,
              "justifyContent": "center",
              "maxWidth": "100%",
              "overflow": "hidden",
            }
          }
        >
          <Text
            ellipsizeMode="tail"
            numberOfLines={1}
            selectable={false}
            style={
              Object {
                "color": "#000000",
                "fontFamily": "GenEiGothicP-SL",
                "fontSize": 17,
                "fontWeight": "400",
                "lineHeight": 25.5,
                "maxWidth": "100%",
                "opacity": 1,
                "paddingBottom": 4,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 4,
                "textAlign": "left",
              }
            }
          >
            
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 56,
            "maxWidth": "100%",
          }
        }
        testID="SubmitButton"
      >
        <View
          size={1}
          style={
            Object {
              "paddingHorizontal": 16,
              "width": "100%",
            }
          }
          title="Create Space"
        >
          <TouchableOpacity
            activeOpacity={1}
            hoverStyle={Object {}}
            onPress={[Function]}
            onPressIn={[Function]}
            onPressOut={[Function]}
          >
            <Text
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Object {
                  "backgroundColor": "#EAEAEA",
                  "borderColor": "#777777",
                  "borderRadius": 7.4375,
                  "borderWidth": 0,
                  "color": "#777777",
                  "fontFamily": "GenEiGothicP-R",
                  "fontSize": 17,
                  "fontWeight": "500",
                  "lineHeight": 42.5,
                  "overflow": "hidden",
                  "paddingHorizontal": 32,
                  "textAlign": "center",
                  "width": "100%",
                }
              }
            >
              Create Space
            </Text>
          </TouchableOpacity>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 44,
            "maxWidth": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "flex": 1,
              "flexDirection": "column",
              "flexGrow": 1,
              "justifyContent": "center",
              "maxWidth": "100%",
              "overflow": "hidden",
            }
          }
        >
          <Text
            ellipsizeMode="tail"
            numberOfLines={1}
            selectable={false}
            style={
              Object {
                "color": "#000000",
                "fontFamily": "GenEiGothicP-SL",
                "fontSize": 17,
                "fontWeight": "400",
                "lineHeight": 25.5,
                "maxWidth": "100%",
                "opacity": 1,
                "paddingBottom": 4,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 4,
                "textAlign": "left",
              }
            }
          >
            
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "maxWidth": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "flex": 1,
              "flexDirection": "column",
              "flexGrow": 1,
              "justifyContent": "center",
              "maxWidth": "100%",
              "overflow": "hidden",
            }
          }
        >
          <Text
            selectable={false}
            style={
              Object {
                "color": "#777777",
                "fontFamily": "GenEiGothicP-SL",
                "fontSize": 14.875,
                "fontWeight": "400",
                "lineHeight": 22.3125,
                "maxWidth": "100%",
                "opacity": 1,
                "paddingBottom": 4,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 4,
                "textAlign": "left",
              }
            }
          >
            Using account: jonsnow@winterfell.com
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "right": 0,
          },
          Object {
            "height": 0,
          },
          undefined,
        ]
      }
    />
  </ScrollView>
</SafeAreaView>
`;
